### DISCUSSION TITLE
Adding musical symbols

### DISCUSSION THREAD

**Discussion by dfballes** [psb1558/Junicode-font#99] [Ideas]
Dear Peter,

is it possible to add musical symbols like accidentals (# is not a problem, but I am missing the flat "b" a lot, and perhaps microtones) to Junicode? I'm using the font since my time at the university and really like it the most because of its shape and all the characters, but at the moment I am teaching music at a school and would really like to have these symbols in my worksheets without changing the font. This always creates a mess in line spacing...

It sure would be too much to ask for musical notes to write rhythms, since Junicode is not a specialized music font, but it would be a nice feature as well ;-) 

Kind regards,
Daniel

**Comment by psb1558** [comment]
Can you post a list of the musical symbols that are necessary for basic coverage, with Unicodes? I take it that you're not interested in actually setting music, but rather handling a few situations that come up in writing about it.

There are musical symbols in the Fell specimen book, but I don't know how suitable they are for modern usage. If you can help track down suitable models, that would also be helpful. I'm afraid that my knowledge of music doesn't extend far beyond knowing how to subscribe to Spotify.

**Comment by dfballes** [comment]
I took the Unicodes from that list: [https://unicode-table.com/en/blocks/musical-symbols/](url):
**Staff and Clefs:**
Five-Line-Staff (1D11A)
G-Clef (1D11E)
C-Clef (1D120)
F-Clef (1D122)
**Accidentals (the first three are the symbols that I'm missing the most at the moment):**
Flat (266D)
Natural (266E)
Sharp (266F)
Double Sharp (1D12A)
Double Flat (1D12B)
**Time Signatures:**
Common Time (1D134)
Cut Time (1D135)
**Rests (1D13B-1D13F):**
Whole Rest
Half Rest
Quarter Rest
Eighth Rest
Sixteenth Rest
**Notes (1D15D-1D161):**
Whole Note
Half Note
Quarter Note
Eighth Note
Sixteenth Note
**Augmentation Dot (1D16D)**

It would be great, if all these Symbols could be implemented in Junicode, and even more: There are Unicodes for medieval music symbols (very rudimentary, but at least something), which would have been great vor my master thesis some years ago... There could be musicologists interested in using this.

The symbols in the list above would be more than sufficient for my daily work, but the absolute best would be, if I could also notate rhythms (like with the Font by Matthew Hindson [https://en.maisfontes.com/rhythms.font](url) or the MakeMusic-Finale-font MaestroTimes, which doesn't care about correct Unicodes for the Symbols (because some don't even exist in Unicode).

**Comment by psb1558** [comment]
I was just surveying the academically oriented fonts I consult most often—for example, Brill, Gentium, SIL Doulos—and couldn't find a single one that includes anything from the Musical Symbols range. This suggests to me that fontmakers think of musical symbols as an issue separate from text, suitable for treatment in separate fonts.

I'm not ruling out the idea of musical symbols completely, but given the amount of work involved and what I've already got on my plate, it's hard to see it happening soon.

I'd probably be receptive if anyone else wanted to take on the work and submit a pull request.

**Reply by dfballes** [reply]
Thank you anyway for your replies, I understand very well that this would be much work.

**Comment by kenmcd** [comment]
@dfballes There are a number of more modern music fonts which will have what you need.
The fonts you mentioned are the old style.
You should probably be using a SMuFL (Standard Music Font Layout) compliant font for text.
The "text" fonts are for use in normal word processors, etc.
For example:
Bravura - is for use in dedicated music writing apps.
Bravura Text - is for use in apps like LibreOffice Writer.

BravuraText has all of that Unicode music block and much more.
https://github.com/steinbergmedia/bravura
Also see these fonts which may or may not have what you need:
Gootville - https://github.com/musescore/MuseScore/tree/master/fonts/gootville/
Leipzig - https://github.com/rism-digital/leipzig/ 
Leland Text - https://github.com/MuseScoreFonts/Leland/
Petaluma - https://github.com/steinbergmedia/petaluma/
Etc., etc. .... all FOSS fonts, all perhaps better suited to your task.

If you have an issue with line spacing, just set a fixed line spacing in your document.
That will override the different vertical metrics in the different fonts.

@psb1558 , SMuFL (Standard Music Font Layout) is like MUFI for music fonts.

**Reply by dfballes** [reply]
Thank you for your hint to fixed line spacing, I never used that feature in Word and think is just the right thing for my purposes. I really like Junicode, that's why I don't want to change to a SMuFL font for text - they often look too much like Times New Roman... But that way I can do all the inserts directly in the text and just have to change the font, without fearing the spacing-mess I normally had. Thank you!

**Comment by psb1558** [comment]
Thanks, @kenmcd. I read about Standard Music Font Layout while looking into this issue. It's big—music notation is complicated!

**Reply by dfballes** [reply]
That is right, if you want to do a font for use in a notation software like Finale or Sibelius. For writing about music, it seems to me to be much less complicated, because normally you need way fewer symbols and possibilities. Except if you want to use Word etc. as a kind of notation software, which is possible, but makes the requirements for a font complicated again (see the fonts by Ansgar Krause [https://www.etsy.com/de/shop/FontsAndMusic](url)
